packages/editor/src/browser/editor.ts

```diff
@@ -81,6 +81,12 @@ export interface TextEditor extends Disposable, TextEditorSelection, Navigatable
     deltaDecorations(params: DeltaDecorationParams): string[];
 
     getVisibleColumn(position: Position): number;
+
+    /**
+     * Replaces the text of source given in ReplacetextParams.
+     * @param params: ReplaceTextParams
+     */
+    replaceText(params: ReplaceTextParams): Promise<boolean>;
```

Comment:
It would be better to have the general solution: https://github.com/theia-ide/theia/issues/1756

Please open a follow-up issue to refactor it.

Reply:
https://github.com/theia-ide/theia/issues/1788